crates/chain/src/lib.rs

```diff
@@ -43,6 +43,8 @@ pub mod tx_graph;
 pub use tx_graph::TxGraph;
 mod chain_oracle;
 pub use chain_oracle::*;
+mod canonical_iter;
+pub use canonical_iter::*;
```

Comment:
We have an iterator `CanonicalIter` which should be public since it's returned by `TxGraph::canonical_iter`. `<CanonicalIter as Iterator>::Item` contains `CanonicalReason` so that needs to be public. Subsequently, `LastSeenIn` also needs to be public.

Otherwise, I've tried to reduce the publicly exposed items in the latest push.